### PR TITLE
[W30] Ray Lin: 7/25 - 7/29

### PR DESCRIPTION
**Pull request by raylin**

### CODE REVIEW

File: RayLin.md [hunk 1]
```diff
@@ -1,12 +1,43 @@
...
+# 07/25 ~ 07/29
 
 - [Video Control]
   - Bug 1271765 - Visual refresh of media controls
-    - UI discussion with Peko
-  - Bug 681553 - [meta] Make HTML5 \<audio\> controls less of a hack
-    - No clear goals for now. Need define what we want controls optimized for audio, and UX should do that defining.
-    - De-prioritized. Save this bug to after visual refresh.
+    - Prefer design 01
+    - Consider special cases:
```
Comment (timdream) [inline RayLin.md]: Are these discussions or patch WIP? Could you upload the patch to the bug to document your progress? thanks!

File: RayLin.md [hunk 2]
```diff
@@ -1,12 +1,43 @@
...
+      - Extreme small size video
+      - Error message refresh
   - Bug 1222273 - Convert HTML \<video\>/\<audio\> controls to be (mostly?) HTML instead of XUL
-    - WIP
-    - All but \<stack\> tag replaced with HTML.
-    - Adjust position: removing XUL specific style
+    - Struggle with mobile UI and XBL inheritance problem.
+    - Conversion progress:
+      - Desktop:
+        - Mostly XUL converted, except \<xul:stack\> and \<xul:scale\>
```
Comment (timdream) [inline RayLin.md]: Maybe we could ask for some guidance from Firefox peers?

File: RayLin.md [hunk 3]
```diff
@@ -1,12 +1,43 @@
...
+        - All functions and interface work fine in normal/large size video.
+        - In small size, e.g. 120px, audio button mis-align...need a subtle CSS adjustment.
+      - Mobile:
+        - Like desktop, xul tag mostly removed
+        - Timeline bar's height could not sized as expected
+        - Abnormal touch event on \<xul:scale\> after de-xul (some hack suppress events in original impl)
+
+### Video Control Planning ###
```
Comment (timdream) [inline RayLin.md]: Nice, @weslyhuang do we need to put this somewhere in Taipei project update?

File: RayLin.md (hunk 2 above)
Comment (timdream) [inline RayLin.md]: What I meant to say is, there are good findings and progress that is worthy of document on the bug itself, not just the weekly here. More people would likely be able to read about it and comment on it (if they could help).

Comment (raylin) [inline RayLin.md]: I'll comment on bug later :)

File: RayLin.md (hunk 1 above)
Comment (raylin) [inline RayLin.md]: I think first patch will be uploaded late this week as Peko might get confirmation this Wed.

**Pull request merged**